src/one_hot_encoding.jl

```diff
@@ -1,40 +1,44 @@
 """
-    OneHotEncoding <: Transform
+    OneHotEncoding{R<:Real} <: Transform
 
 One-hot encode the categorical value for each target element.
 
 Construct a n-by-p binary matrix, given a `Vector` of target data `x` (of length n) and a
 `Vector` of all unique possible values in x (of length p).
 
-The element [i, j] is `1` if the i^th target in `x` corresponds to the j^th possible
-value and `0` otherwise.
+The element [i, j] is `true` if the i^th target in `x` corresponds to the j^th possible
+value and `false` otherwise. Note that `R`can be specified to determine the return type
+of results. It defaults to a `Matrix` of `Bool`s.
 
 Note that this Transform does not support specifying dims other than `:` (all dims) because
 it is a one-to-many transform (for example a `Vector` input produces a `Matrix` output).
 """
-struct OneHotEncoding <: Transform
-    categories::Dict{Any, Int}
+struct OneHotEncoding{R<:Real, T} <: Transform
```

Comment:
do we need to parameterise this by `T` as well? It doesn't seem to be used for anything - and we don't mention it in the docstring.

Reply:
It's used for space efficiency so we don't need to store a Dict of `Any`